build.rs

```diff
@@ -1,50 +1,56 @@
+#[cfg(feature = "bindgen")]
 extern crate bindgen;
-extern crate skeptic;
+
+use std::env;
+use std::fs;
 use std::path::PathBuf;
 use std::process::Command;
-use std::env;
-
 
 fn main() {
+    Command::new("make")
+        .current_dir("bwa")
+        .arg("CFLAGS=-g -Wall -O2 -fPIC")
+        .arg("libbwa.a")
+        .status()
+        .ok()
+        .expect("Failed to build bwa");
 
-    Command::new("make").current_dir("bwa")
-                        .arg("CFLAGS=-g -Wall -O2 -fPIC")
-                        .arg("libbwa.a")
-                        .status().ok().expect("Failed to build bwa");
-
-    
     println!("cargo:rustc-link-search=bwa");
     println!("cargo:rustc-link-lib=static=bwa");
-
+    let out_path = PathBuf::from(env::var("OUT_DIR").unwrap());
     // The bindgen::Builder is the main entry point
     // to bindgen, and lets you build up options for
     // the resulting bindings.
-    let bindings = bindgen::Builder::default()
-        // The input header we would like to generate
-        // bindings for.
-        .header("wrapper.h")
-        .whitelist_function("mem_align1_core")
-        .whitelist_function("mem_sam_pe")
-        .whitelist_function("mem_opt_init")
-        .whitelist_function("bwa_idx_load")
-        .whitelist_function("bwa_idx_destroy")
-        .whitelist_function("mem_process_seq_pe")
-        .whitelist_function("bwa_fill_scmat")
-        .whitelist_var("BWA_IDX_*")
-
-        // Finish the builder and generate the bindings.
-        .generate()
-        // Unwrap the Result and panic on failure.
-        .expect("Unable to generate bindings");
-
-    // Write the bindings to the $OUT_DIR/bindings.rs file.
-    let out_path = PathBuf::from(env::var("OUT_DIR").unwrap());
-    bindings
-        .write_to_file(out_path.join("bindings.rs"))
-        .expect("Couldn't write bindings!");
-
-
-
-    // generates doc tests for `README.md`.
-    // skeptic::generate_doc_tests(&["README.md"]);
+    // If bindgen is enabled, use it
+    #[cfg(feature = "bindgen")]
```

Comment:
you can only actually switch this on if you put a `bindgen` feature in the Cargo.toml.  My recommendation would be to remove this block & put a description of how to manually generate the bindings:

```cargo install bindgen
bindgen --no-doc-comments --whitelist-function mem_align1_core <more args mirroring this section> wrapper.h -o <appropriate bindings file>
```

We should probably also include a Mac version for completeness (or test that the generated files work on Mac)

Reply:
I did not realize you could do this from the command line. Yes, will get rid of this.